contracts/Strategy.sol

```diff
@@ -72,24 +143,79 @@ contract Strategy is BaseStrategy {
         // NOTE: Maintain invariant `want.balanceOf(this) >= _liquidatedAmount`
         // NOTE: Maintain invariant `_liquidatedAmount + _loss <= _amountNeeded`
 
-        uint256 totalAssets = want.balanceOf(address(this));
-        if (_amountNeeded > totalAssets) {
-            _liquidatedAmount = totalAssets;
-            _loss = _amountNeeded.sub(totalAssets);
+        uint256 _liquidWant = wantBalance();
+
+        _amountNeeded = Math.min(_amountNeeded, estimatedTotalAssets()); // Otherwise we can end up declaring a liquidation loss when _amountNeeded is more than we own
+
+        if (_liquidWant < _amountNeeded) {
+            _unstakeLPTokens(stakedLPBalance());
```

Comment:
why unstake the full amount when we can compute the proper amount and withdraw only what we need?

Reply:
I guess I wasn't quite sure about how to deal with unexpected losses on withdraws. I'll try to think of a better solution